network/netplay/netplay_discovery.c

```diff
@@ -234,57 +255,102 @@ bool netplay_lan_ad_server(netplay_t *netplay)
-            NETPLAY_HOST_STR_LEN);
-         strlcpy(ad_packet_buffer.content, !string_is_empty(
-                  path_basename(path_get(RARCH_PATH_BASENAME))) 
-               ? path_basename(path_get(RARCH_PATH_BASENAME)) : "N/A",
-               NETPLAY_HOST_LONGSTR_LEN);
-         strlcpy(ad_packet_buffer.nick, netplay->nick, NETPLAY_HOST_STR_LEN);
+         if (!net_ifinfo_new(&interfaces))
+            return false;
 
-         if (info)
+         for (int k = 0; k < interfaces.size; k++)
          {
-            strlcpy(ad_packet_buffer.core, info->info.library_name,
-               NETPLAY_HOST_STR_LEN);
-            strlcpy(ad_packet_buffer.core_version, info->info.library_version,
-               NETPLAY_HOST_STR_LEN);
+            char *p;
+            char sub[NETPLAY_HOST_STR_LEN];
+
+            p=strrchr(reply_addr,'.');
+            if (p)
+            {
+               strlcpy(sub, reply_addr, p - reply_addr + 1);
+               if (strstr(interfaces.entries[k].host, sub) && !strstr(interfaces.entries[k].host, "127.0.0.1"))
+               {
+                  RARCH_LOG ("[discovery] query received on common interface: %s/%s (theirs / ours) \n", reply_addr, interfaces.entries[k].host);
+
+                  info = runloop_get_system_info();
+
+                  /* Now build our response */
+                  content_crc = content_get_crc();
+
+                  memset(&ad_packet_buffer, 0, sizeof(struct ad_packet));
+                  memcpy(&ad_packet_buffer, "RANS", 4);
+
+                  strlcpy(ad_packet_buffer.address, interfaces.entries[k].host,
+                     NETPLAY_HOST_STR_LEN);
+                  ad_packet_buffer.protocol_version =
+                     htonl(NETPLAY_PROTOCOL_VERSION);
+                  ad_packet_buffer.port = htonl(netplay->tcp_port);
+                  strlcpy(ad_packet_buffer.retroarch_version, PACKAGE_VERSION,
+                     NETPLAY_HOST_STR_LEN);
+                  strlcpy(ad_packet_buffer.content, !string_is_empty(
+                           path_basename(path_get(RARCH_PATH_BASENAME))) 
+                        ? path_basename(path_get(RARCH_PATH_BASENAME)) : "N/A",
+                        NETPLAY_HOST_LONGSTR_LEN);
+                  strlcpy(ad_packet_buffer.nick, netplay->nick, NETPLAY_HOST_STR_LEN);
+
+                  if (info)
+                  {
+                     strlcpy(ad_packet_buffer.core, info->info.library_name,
+                        NETPLAY_HOST_STR_LEN);
+                     strlcpy(ad_packet_buffer.core_version, info->info.library_version,
+                        NETPLAY_HOST_STR_LEN);
+                  }
+
+                  snprintf(s, sizeof(s), "%d", content_crc);
+                  strlcpy(ad_packet_buffer.content_crc, s,
+                     NETPLAY_HOST_STR_LEN);
+
+                  /* Build up the destination address*/
+                  struct addrinfo hints = {0}, *addr;
+                  char port_str[6];
```

Comment:
C89 errors

Reply:
what's the error here? I'm not familiar with C89 restrictions.

Reply:
NVM